static/js/util/profile_edit.js

```diff
@@ -81,3 +104,44 @@ export function saveAndContinue(next) {
     browserHistory.push(next)
   });
 }
+
+export function editProfileObjectArray (arrayName, blankEntry, formCallback) {
+  const { updateProfile, profile } = this.props;
+  if ( profile[arrayName] ) {
+    let editForms = profile[arrayName].map( (entry, index) => (
+      formCallback.bind(this)(entry, index)
```

Comment:
If you make that change to `formCallback` you should be able to simplify this to

    let editForms = profile[arrayName].map(formCallback);